### PR TITLE
Fix remote integration tests

### PR DESCRIPTION
**Pull request by troychiu**
# TL;DR
Currently, the integration tests of remote is not in use, and the test files are out of date. I clean them up and reactivate it in the workflow.

## Type
 - [ ] Bug Fix
 - [X] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
1. Use the same idea as eager workflow tests to register workflows before testing.
2. Remove legacy tests.
3. Fix tests that will fail.
## Tracking Issue
https://github.com/flyteorg/flyte/issues/4250

## Follow-up issue
Increase the code coverage

### CODE REVIEW

**Comment by samhita-alla** [comment]
@troychiu, is this still WIP?

**Comment by troychiu** [comment]
Hi @samhita-alla, I think this pr has been done and all the tests will pass in local environment. However, all the tests will fail in CI pipeline. I am not sure if the error comes from any issue of the CI environment. @samhita-alla @pingsutw, do you have any idea? Thank you,.

**Comment by eapolinario** [comment]
@troychiu , I [updated](https://github.com/flyteorg/flytekit/pull/1896/commits/ebf91d8bda04d2013808e3830e1b5e80e7f8948a) your PR to use the right config file in CI (and locally).

**Comment by troychiu** [comment]
@eapolinario thank you so much. Let me resolve the fail tests.

**Comment by eapolinario** [comment]
@troychiu , I pushed another change to skip the failing test for the time being. We should be able to re-enable it shortly, but I don't want to block your PR on this.

**Comment by eapolinario** [comment]
@troychiu , tests should pass now. I merged master (which contained an unrelated fix for an issue in `tracker.py`) and skipped the 2 failing tests. I tested the change I mentioned in the reason locally just to be safe and tests pass. I'll follow up in a separate PR.

**Review by eapolinario — APPROVED** [review]

**Comment by troychiu** [comment]
Thank you for the help @eapolinario !
